### PR TITLE
[GOBBLIN-1657] Update completion watermark on change_property in IcebergMetadataWriter

### PR DESCRIPTION
**Pull request by vikrambohra**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [ ] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-1657] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1657

### Description
- [ ] Here are some details about my PR, including screenshots (if applicable):
    - There are quiet topics where data can be generated quiet infrequently causing the completion watermark to lag. Also, a commit operation could miss completion watermark update if there is a lag in kafka audit.

To fix above issues:
- Added logic to check audit count on change_property operation type for the next window of current watermark 
If counts match or an empty map is returned then consider audit count as complete and update the watermark.
- Added a config in KafkaAuditVerifier to return complete if found an empty map (with no counts from audit system) 

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
  Added unit test to update completion watermark on change_property

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gobblin-iceberg/src/main/java/org/apache/gobblin/iceberg/writer/IcebergMetadataWriter.java [hunk 1]
```diff
@@ -790,28 +789,33 @@ public void flush(String dbName, String tableName) throws IOException {
...
-              }
-            }
+            checkAndUpdateCompletenessWatermark(tableMetadata, topic, tableMetadata.datePartitions, props);
           }
         }
         if (tableMetadata.deleteFiles.isPresent()) {
           tableMetadata.deleteFiles.get().commit();
         }
+        // Check and update completion watermark when there are no files to be registered, typically for quiet topics
+        // The logic is to check the next window from previous completion watermark and update the watermark if there are no audit counts
```
Comment (sv2000) [inline gobblin-iceberg/src/main/java/org/apache/gobblin/iceberg/writer/IcebergMetadataWriter.java:803]: Is the next window the starting time for completeness checks or will we only check for the completeness of the next window?

**Review by sv2000 — CHANGES_REQUESTED** [review]

**Comment by sv2000** [comment]
@vikrambohra Also check for travis build failure.

File: gobblin-iceberg/src/main/java/org/apache/gobblin/iceberg/writer/IcebergMetadataWriter.java (hunk 1 above)
Comment (vikrambohra) [reply gobblin-iceberg/src/main/java/org/apache/gobblin/iceberg/writer/IcebergMetadataWriter.java:803]: Just the next window =  [currentWatermark, currentWatermark + 1 hour]. However audit service will not be called if this window falls in the current hour.

File: gobblin-completeness/src/main/java/org/apache/gobblin/completeness/verifier/KafkaAuditCountVerifier.java [hunk 2]
```diff
@@ -45,6 +45,8 @@ public class KafkaAuditCountVerifier {
   public static final String REFERENCE_TIERS = COMPLETENESS_PREFIX + "reference.tiers";
   public static final String THRESHOLD = COMPLETENESS_PREFIX + "threshold";
   private static final double DEFAULT_THRESHOLD = 0.999;
+  public static final String COMPLETE_ON_NO_COUNTS = COMPLETENESS_PREFIX + "complete.on.no.counts";
```
[the remark this answers is not in the document]
Comment (vikrambohra) [reply gobblin-completeness/src/main/java/org/apache/gobblin/completeness/verifier/KafkaAuditCountVerifier.java:48]: Default behavior is for other users of this class to not want a "success" if there is an empty response from kafka audit.

This config is only for this use case where we want a "success" on empty response from kafka audit.

File: gobblin-iceberg/src/main/java/org/apache/gobblin/iceberg/writer/IcebergMetadataWriter.java (hunk 1 above)
Comment (sv2000) [reply gobblin-iceberg/src/main/java/org/apache/gobblin/iceberg/writer/IcebergMetadataWriter.java:803]: So what happens if the current time is 08:00 and current watermark is 02:00? Will we only check for the completeness for [02:00, 03:00]? Shouldn't we check for completeness of the entire range [02:00, 08:00]?

Comment (vikrambohra) [reply gobblin-iceberg/src/main/java/org/apache/gobblin/iceberg/writer/IcebergMetadataWriter.java:803]: This case should happen only after a long downtime. Even then watermark should catchup quickly as we will be flushing every minute. And if there is a append file operation in between then we would catchup more aggressively

In regular scenario. We will update watermark on the next flush once current time is the next hour of current watermark.

**Review by sv2000 — APPROVED** [review]
+1. LGTM.

**Pull request merged**